### PR TITLE
Docker: Improve Published Docker Images

### PR DESCRIPTION
**Pull request by taion809**
*title*: Improve Published Docker Images
This is an initial update to the docker images that are created
and published to hub.dockerhub.com.

The purpose is to provide users with an image that can be ready
to run from the moment they issue docker pull.

ref: #2805 

Signed-off-by: Nicholas Johns <nicholas.a.johns5@gmail.com>

### CODE REVIEW

**Comment by zuercher** [comment]
@junr03 can you take the first pass on this?

File: ci/Dockerfile-envoy-alpine-debug [hunk 1]
```diff
@@ -1,3 +1,17 @@
 FROM frolvlad/alpine-glibc
 
+ENV loglevel=info
```
Comment (junr03) [inline ci/Dockerfile-envoy-alpine-debug]: I think this should be debug

File: ci/Dockerfile-envoy-image [hunk 2]
```diff
@@ -1,3 +1,26 @@
 FROM ubuntu:16.04
 
+ARG diversion=1.2.1
```
Comment (junr03) [inline ci/Dockerfile-envoy-image]: can we call this `dumbinit_version` or something similar?

**Comment by taion809** [comment]
> General question: now the published images will be default run with the config you specified above, what is the process for users to override that? Would the build FROM this image and override what gets mounted and the CMD?

There are 2 ways to override the config.  You can extend the image by building a new image based off `envoyproxy/envoy` and copying the new config to `/etc/envoy/envoy.yaml`.  It is also possible to use a volume to mount a configuration at that location (useful when developing and testing new configs quickly)

Example `docker-compose.yml`:
```yaml
version: '3'
services:
  envoy:
    image: envoyproxy/envoy:latest
    ports:
      - "8080:10000"
    volumes:
      - ./envoy.yaml:/etc/envoy/envoy.yaml
```

File: ci/Dockerfile-envoy-alpine-debug (hunk 1 above)
Comment (taion809) [reply ci/Dockerfile-envoy-alpine-debug]: i forgot to mention, this can be overridden at runtime, `docker run -e "loglevel=debug" envoyproxy/envoy`

**Comment by taion809** [comment]
@junr03 can you take another look, I've provided some updates to the data-plane-api documentation (referenced above).

**Comment by junr03** [comment]
@taion809 are you done committing? Let me know when you are ready, and I will take another look.

**Comment by taion809** [comment]
@junr03 yeah, sorry there was a last minute issue that i fixed (docker interprets CMD differently based on whether or not you've used an array or a flat string :|)

**Comment by taion809** [comment]
@junr03 looks like all the checks passed so i think it's ready for you to take another look

File: ci/Dockerfile-envoy-alpine-debug (hunk 1 above)
Comment (junr03) [reply ci/Dockerfile-envoy-alpine-debug]: Yes, debug logging is pretty expensive. Ok that sounds good to me

**Pull request merged**